scvi/model/base/_demixin.py

```diff
@@ -0,0 +1,453 @@
+import logging
+from functools import partial
+from typing import Iterable, Optional, Sequence, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+from sklearn.covariance import EllipticEnvelope
+from torch.distributions import Categorical, Normal
+
+from scvi._compat import Literal
+from scvi._docs import doc_differential_expression
+from scvi._utils import _doc_params
+from scvi.model._utils import (
+    _get_batch_code_from_category,
+    _get_var_names_from_setup_anndata,
+    scrna_raw_counts_properties,
+)
+from scvi.model.base._utils import _de_core
+
+logger = logging.getLogger(__name__)
+Number = Union[int, float]
+
+
+class DEMixin:
+    """
+    DE module relying on Importance-sampling.
+
+    Mixin for using
+    importance-weighted DE content.
+    This however requires some additional structure on the
+    module's (e.g., VAE) methods and associate signatures
+    """
+
+    @_doc_params(
+        doc_differential_expression=doc_differential_expression,
+    )
+    def lvm_de(
```

Comment:
can we somehow call this `differential_expression` and add a mode to it for importance weighted estimator and then remove the DE function out of `RNAMixin`?

Reply:
I see, the idea would be to move `get_normalized_expression` and `differential_expression` from `_rnamixin.py`, and compartmentalize every feature for differential expression (plugin and IS alike) in `_demixin.py`?
I am getting this right?

Reply:
Yes, but keep `get_normalized_expression` in RNA mixin, just move all DE specific functionality into this mixin. Though I see now it can get messy since `differential_expression` assumes `get_normalized_expression` exists...

Reply:
do you think there's a way around that?

Reply:
@adamgayoso  I found a solution that I find pretty attractive 5f5ac98.
However, the new signature of `differential_expression` is even more intricate than it was before.
I am afraid that the `differential_expression` function is becoming a bit too heavy.
I think that now (or in a following up PR) is a good time to think about ways to facilitate and clarify DE queries within `scvi-tools`.
Do you agree?

Here are some random ideas that come to mind, I don't really know if they are good ones:
1. **Splitting up how DE is performed in two steps**. 
I think that performing a significant number of subpopulation comparisons within an experiment is a common use case. Unfortunately, the current codebase requires to specify which test is applied each time `differential_expression` is called, which is pretty cumbersome in my opinion.
For this reason, an option could be to split up `differential_expression` arguments into two categories: the ones that define which test to apply, and the ones that characterize which subpopulations are compared.


We would expect the arguments of the first category to be applied only once, before any DE query, but we would obviously let users change those parameters at any point.
In the first step, the user would explicit.

```python
    model = SCVI(adata=adata)
    model.train()

    # Step 1: specify which test is applied
    ## either 
    model.setup_differential_expression()  # default mode
    ## or
    model.setup_differential_expression(importance_sampling=True, delta=None, fdr_target=0.1)
    
    # Step 2: performing interactive DE queries
    results_1 = model.differential_expression(
        idx1="(labels == 'label_1') & (_scvi_batch == 1)",
        idx1="(labels == 'label_1') & (_scvi_batch == 0)",
        batch_correction=False,
    )
    results_2 = model.differential_expression(
        idx1="(labels == 'label_1')",
        idx1="(labels == 'label_2')",
        batch_correction=False,
    )
```

2. **Visualizing which test and which subpopulations are compared**
In a similar fashion than `setup_anndata` within a notebook, `setup_differential_expression` could display which test to perform, clarifying what is happening under the hood.
We could also do a similar thing when `differential_expression` is called, but I am not sure this is as important